apps/gallery/js/thumbnail_date_group.js

```diff
@@ -59,7 +60,8 @@ function ThumbnailDateGroup(item) {
   this.htmlNode.appendChild(this.header);
   this.htmlNode.appendChild(this.container);
 
-  this.localize();
+  this.header.textContent =
+    IntlHelper.get('date-group').format(new Date(this.date));
```

Comment:
Why can't you just call localize() here?  The thumbnails array is empty here in the constructor, so localize() will just do the same as this code, won't it?  I think it would be cleaner to do all localization in the localize method